app/src/main/java/io/appium/uiautomator2/handler/Getpackages.java

```diff
@@ -0,0 +1,47 @@
+package io.appium.uiautomator2.handler;
+
+
+import static io.appium.uiautomator2.model.Session.NO_ID;
+import static io.appium.uiautomator2.server.ServerInstrumentation.ServerContext;
+
+import android.app.Activity;
+import android.content.pm.ApplicationInfo;
+import android.content.pm.PackageManager;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.appium.uiautomator2.handler.request.NoSessionCommandHandler;
+import io.appium.uiautomator2.handler.request.SafeRequestHandler;
+import io.appium.uiautomator2.http.AppiumResponse;
+import io.appium.uiautomator2.http.IHttpRequest;
+import io.appium.uiautomator2.model.api.BatteryStatusModel;
+import io.appium.uiautomator2.model.api.StatusModel;
+import io.appium.uiautomator2.model.api.touch.PackageModel;
+import io.appium.uiautomator2.server.ServerInstrumentation;
+import io.appium.uiautomator2.utils.BatteryHelper;
+
+public class Getpackages extends SafeRequestHandler implements NoSessionCommandHandler {
+    public Getpackages(String mappedUri) {
+        super(mappedUri);
+    }
+    public static List<PackageModel> appDetails = new ArrayList<PackageModel>();
+    @Override
+    protected AppiumResponse safeHandle(IHttpRequest request) {
+        try{
+            List<ApplicationInfo> apps = ServerContext.getPackageManager().getInstalledApplications(PackageManager.GET_META_DATA);
+            for(ApplicationInfo appInfo: apps) {
+                PackageModel model = new PackageModel();
+                if(ServerContext.getPackageManager().getLaunchIntentForPackage(appInfo.packageName)!=null) {
+                    model.setPackageName(appInfo.packageName);
+                    model.setAppName((String) ServerContext.getPackageManager().getApplicationLabel(appInfo));
+                    model.setPackageActivity(ServerContext.getPackageManager().getLaunchIntentForPackage(appInfo.packageName).getComponent().getClassName());
```

Comment:
this line is too long

Reply:
Yes possibly, i will do the changes thanks

Reply:
The return value description from the documentation of `getLaunchIntentForPackage` is:

> The current implementation looks first for a main activity in the category [Intent#CATEGORY_INFO](https://developer.android.com/reference/android/content/Intent#CATEGORY_INFO), and next for a main activity in the category [Intent#CATEGORY_LAUNCHER](https://developer.android.com/reference/android/content/Intent#CATEGORY_LAUNCHER).

So, maybe the name should be `launcherActivityName` or similar to indicate it is just for possible launcher activity.

Btw, have you tried `getLaunchIntentSenderForPackage`?  https://developer.android.com/reference/android/content/pm/PackageManager#getLaunchIntentSenderForPackage(java.lang.String)
I wonder if the api is proper for android sdk level 33+

Reply:
Thanks for the suggestion, let me check, i will also try to check in almost all SDKs using emulators

Reply:
I have tried <getLaunchIntentSenderForPackage> but it's not giving me the component info but <getLaunchIntentForPackage> working perfectly

Reply:
thanks for the checking